optimum/habana/transformers/generation/utils.py

```diff
@@ -542,6 +542,9 @@ def generate(
             generation_config.ignore_eos = kwargs.get("ignore_eos", lazy_mode)
         generation_config.validate()
         model_kwargs = generation_config.update(**kwargs)  # All unused kwargs must be model kwargs
+        if self.config.model_type == "falcon" and "token_type_ids" in kwargs.keys():
+            for key in ["token_type_ids"]:
```

Comment:
Should this be :
```suggestion
            for key in kwargs["token_type_ids"]:
```

Reply:
I based my changes off https://github.com/HabanaAI/optimum-habana-fork/pull/15 so I think this code is meant to support falcon-180b inference. It is not relevant for the change I'd like to make.

Reply:
@tjs-intel , for any falcon-180b inference related code change, please work with @schoi-habana to submit PR to internal oh-fork repo.

Reply:
I have done the opposite and removed the code change related to falcon-180b